### PR TITLE
rename pod injection 

### PR DESCRIPTION
**Pull request by jessfraz**
from https://github.com/kubernetes/community/pull/254#issuecomment-282050817

we also need to decide on the api group

ping @pmorie @erictune @bgrant0607

### CODE REVIEW

**Comment by pmorie** [comment]
There seem to be two issue to settle:

1.  The name of the API group
2.  The name of the resource

I agree that the 'policy' name should be about what is allowed to do a thing.  By that measure, this is not a policy at all.  I like 'defaults' as a name for this type of resource.  How about this:

- API group: `defaults`
- Resource name: `PodServiceDefault`

**Comment by pmorie** [comment]
cc @pwittrock @smarterclayton @derekwaynecarr

**Comment by derekwaynecarr** [comment]
I also heard negative responses to the Policy suffix when discussing this
today with folks.  Defaults is preferred terminology.

In a side note, LimitRange currently straddles a Policy (says no because of
min or max constraints) and Defaults.  In retrospect having them combined
is useful for cluster operators that want to enforce a default is applied,
but inhibits users from creating their own defaults based on our typical
RBAC setups.   I think something like ResourceDefaults would make a lot of
sense in the future as well and the name fits there for me too.

One concern I have is around API group conventions.  If in 1.6 we add this
resource as v1alpha1 and in 1.7 we desire to move it to v1beta1. Are we
able to add a new *Defaults resource to this group in just the v1alpha1
version?  I am admittedly confused on wheee we stand with mixed sets of
resources across versions in a group.


On Thu, Feb 23, 2017 at 9:33 PM Paul Morie <notifications@github.com> wrote:

cc @pwittrock <https://github.com/pwittrock> @smarterclayton
<https://github.com/smarterclayton> @derekwaynecarr
<https://github.com/derekwaynecarr>

—
You are receiving this because you were mentioned.


Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/community/pull/396#issuecomment-282186773>,
or mute the thread

**Comment by derekwaynecarr** [comment]
settings is fine too but I think injection should be avoided in favor of defaults.

**Comment by derekwaynecarr** [comment]
How about API group settings and resource name is PodServiceDefaults?   I think having the noun associated with the resource it mutates on admission front and center will avoid user confusion.  Having Service* kind of obfuscates that it impacts Pods and not Services.

**Comment by derekwaynecarr** [comment]
saying kubectl get settings/podservicedefaults is not the worst thing...

**Comment by pmorie** [comment]
I can live with `PodServiceDefaults`.

How about you @jessfraz ?

**Comment by liggitt** [comment]
Remember that groups get `k8s.io` suffix. kubectl will find specified resources via server-specified discovery order and do fuzzy matching on group names, so any of the following would work:

```
kubectl get podservicedefaults
kubectl get podservicedefaults.settings
kubectl get podservicedefaults.settings.k8s.io
```

**Comment by bgrant0607** [comment]
@jessfraz Sorry I wasn't clear. I'd like to nix "Service", "Injection", and "Policy".

I like "settings" for the name of the API group.

"Defaults" doesn't quite fit, since the values in the Pod don't gracefully override the values in this resource, according to the proposal.

PodPresets?

@derekwaynecarr Is there a reason you wouldn't add compute resources to this API?

**Comment by jessfraz** [comment]
settings seems ok to me

is pod presets ok? i'll wait for status quo before doing find replace lol

**Comment by pwittrock** [comment]
@bgrant0607 WDYT of "Dependencies" instead of "Defaults"?

**Comment by pmorie** [comment]
I think `settings` is best for the name of the group, and `PodPreset` is fine for the name of the resource.  It's hard to find a perfect name that will be perfect forever; I think `settings/PodPreset` is plenty-good.

**Comment by bgrant0607** [comment]
+1 to `settings/PodPreset`.

@pwittrock While this is similar in some ways to dependency injection, the mechanism is (deliberately) more general purpose than that. For example, it could be used just to set an environment variable in a group of pods.

**Comment by jessfraz** [comment]
updated

File: contributors/design-proposals/pod-preset.md [hunk 1]
```diff
@@ -111,24 +111,24 @@ with the credential and endpoint data required to do so).
...
+// PodPreset is a policy resource that defines additional runtime
 // requirements for a Pod.
-type PodInjectionPolicy struct {
+type PodPreset struct {
     unversioned.TypeMeta
     ObjectMeta
 
     // +optional
-    Spec PodInjectionPolicySpec
+    Spec Podpresetpec
```
Comment (pmorie) [inline contributors/design-proposals/pod-preset.md]: `PodPresetSpec`, plz

Comment (jessfraz) [inline contributors/design-proposals/pod-preset.md]: lol

**Comment by pmorie** [comment]
LGTM - I'll give it 30 min and then merge if no one objects.

**Comment by pwittrock** [comment]
SGTM

**Comment by derekwaynecarr** [comment]
That works for me

**Comment by pmorie** [comment]
Merging, thanks for tolerating the bikeshedding, @jessfraz

**Comment by pmorie** [comment]
@smarterclayton did you want to change the name?

**Comment by smarterclayton** [comment]
I'm trying to reconcile "settings" to how we'd present this in UIs and in relation to other existing groups.  Not opposed to it, although as a group it might be too close to "extensions" (grab bag for my stuff).  I'll squawk if I can come up with an objection.

**Comment by smarterclayton** [comment]
No I decided i liked PodPreset because someone unprompted called it a
preset in describing the concept.

**Pull request merged**